Mathlib/Logic/UnivLE.lean

```diff
@@ -0,0 +1,49 @@
+/-
+Copyright (c) 2023 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import Mathlib.Logic.Small.Basic
+
+/-!
+# UnivLE
+
+A proposition expressing a universe inequality. `UnivLE.{u, v}` expresses that `u ≤ v`,
+in the form `∀ α : Type max u v, Small.{v} α`.
+
+See the doc-string for the comparison with an alternative weaker definition.
+
+See also `Mathlib.CategoryTheory.UnivLE` for the statement
+`UnivLE.{u,v} ↔ EssSurj (uliftFunctor : Type v ⥤ Type max u v)`.
+-/
+
+noncomputable section
+
+/--
+A class expressing a universe inequality. `UnivLE.{u, v}` expresses that `u ≤ v`.
+
+There are (at least) two plausible definitions for `u ≤ v`:
+* strong: `∀ α : Type max u v, Small.{v} α`
+* weak: `∀ α : Type u, Small.{v} α`
+
+The weak definition has the advantage of being transitive.
+However only under the strong definition do we have `Small.{v} ((α : Type u) → (β : Type v))`,
```

Comment:
This is actually a lie and the instance `Small.{v} ((α : Type u) → (β : Type v))` can already be [inferred ](https://github.com/leanprover-community/mathlib4/blob/fbafd41e483ba86bbec53622d589980630ef7bba/Mathlib/Logic/UnivLE.lean#L57) via `small_Pi`.

I started the PR #7695 to redefine `UnivLE`. Actually, I think we do even need to use the fact that `α → β` because we can "shrink" (actually expand) `α` to the larger universe `Type v` right away: if `u ≤ v` then limits indexed by a category in `Type u` can also be indexed by an [equivalent](https://github.com/leanprover-community/mathlib4/blob/fbafd41e483ba86bbec53622d589980630ef7bba/Mathlib/CategoryTheory/EssentiallySmall.lean#L167) category in `Type v`, so if a category has limits of size `v` it also has limits of size `u`, and the same for PreservesLimitsOfSize, etc. I believe changing the indexing category also gives more desirable defeq than shrinking the limit cone point, and allows concentrating attention at the single place (the indexing category).

My PR doesn't yet pass CI and you're all welcome to fix it. Ping: @kbuzzard @joelriou

Reply:
I think "incorrect" is probably better than "a lie". :-) I didn't intend to mislead!

I'm glad you're cleaning this up.

Reply:
Sorry, I didn't meant to accuse you! I've seen the word [used](https://github.com/leanprover-community/mathlib4/commit/c289c2cc94dc6dfd36178985a3aba036a53c8e8d) in an unconventional sense, and thought that sounds funny :)

Reply:
I should add that this is surely not a trivial observation; I've already come across the docstring more than one week ago, but only yesterday I realized it's wrong.

Reply:
No, I had missed this, and believed what I said! :-)